hibernate-envers/src/main/java/org/hibernate/envers/query/internal/impl/RevisionsOfEntityQuery.java

```diff
@@ -71,6 +74,36 @@ public RevisionsOfEntityQuery(
 		this.includePropertyChanges = includePropertyChanges;
 	}
 
+	@Override
+	public AuditAssociationQuery<? extends AuditQuery> traverseRelation(
+			String associationName,
+			JoinType joinType,
+			String alias) {
+
+		if ( !selectEntitiesOnly ) {
+			throw new IllegalStateException( "Audit association queries only permitted while selecting only entities." );
```

Comment:
I don't know too much about the logic requirement, but `while selecting only entities.` seems a little bit not natural.
Either `while selecting entities only.` or `while only selecting entities.` is a little bit better.
Ignore my comment from someone to whom English is not first language.

Reply:
Good suggestion.

Reply:
thanks for adopting. Super happy to review your PR, though I seldom approve it for lack of domain knowledge.